cli-dotnet-test.js

```diff
@@ -22,15 +25,68 @@ const { spawn }   = require("child_process");
 // #region Commands
 
 const dotnetTest = {
-    cmds: {
-        dotnetTest:       "dotnet test --no-build --no-restore",
-        dotnetTestFilter: "dotnet test --no-build --no-restore --filter",
-    },
+    cmd: "dotnet test --no-build --no-restore",
     descriptionSkipClean() {
         return "Skips the clean, build, and restore steps before running the dotnet test runner. This will speed up sequential runs if intentionally running on the same assemblies.";
     },
     description() {
-        return `Runs dotnet test runner on the ${dotnetPath.solutionPath()} solution (via ${this.cmds.dotnetTest})`;
+        return `Runs dotnet test runner on the ${dotnetPath.solutionPath()} solution (via ${this.cmd})`;
+    },
+    runByProject(skipClean) {
+        // Check for the solution path before attempting any work
+        dotnetPath.solutionPathOrExit();
+
+        if (!skipClean) {
+            dotnetBuild.run(true, true);
+        }
+
+        const solutionDir = dotnetPath.solutionDir();
+        dir.pushd(solutionDir);
+
+        const testProjects = shell.find("**/*.Test*.csproj");
+        if (testProjects == null || testProjects.length === 0) {
+            echo.error("Could not find any csproj files matching the pattern *.Test*.csproj.");
+            shell.exit(1);
+        }
+
+        echo.message(`Found ${testProjects.length} test projects in the ${dotnetPath.solutionDir()} solution...`);
+
+        // Since the spawnSync function takes the base command and all arguments separately, we cannot
+        // leverage the base dotnet test command string here. We'll build out the arg list in an array.
+        let cmd = "dotnet";
+
+        testProjects.map((project) => {
```

Comment:
Believe this should be a `forEach`, not `map`

Reply:
If I'm reading this correctly, this will fail the overall process and short-curcuit once one fails. If that is the case, we should change it to collect the results from each test project and then exit. If one or more failed, exit failed and output a summary of each failed project.

Reply:
That's a good point, I didn't consider that. Do you think we'd want the ability to toggle a short-circuit or just always have it run the full suite before determining the failure?

Reply:
While there could be a scenario where someone would want to exit upon an immediate failure, I've never found it. If its a quick add, go for it, but the default behavior should be to run them all and collect all success and failures before exiting.